### ISSUE TITLE
Neovim is installed twice instead upgraded

### ISSUE THREAD

**Issue by jfernandz** [microsoft/winget-cli#4571]
### Brief description of your issue

I've run `winget upgrade --all` and it has installed the new version of Neovim (0.10.0) in parallel to the older one (0.9.5). Now I cannot uninstall any version even using the `-v` argument for `winget uninstall`. 

Also ... not sure why despite my system language is english ... winget still prints its outputs in spanish 🤔 

### Steps to reproduce

Just running `winget upgrade --all`

### Expected behavior

normal upgrade

### Actual behavior

Neovim 0.10.0 has been installed but not upgraded 0.9.5 version, now I cannot remove any one.

```
> winget list neovim
Nombre Id            Versión Disponible Origen
----------------------------------------------
Neovim Neovim.Neovim 0.9.5   0.10.0     winget
Neovim Neovim.Neovim 0.10.0             winget
> winget uninstall Neovim -v 0.9.5
Se encontraron varios paquetes instalados que coinciden con los criterios de entrada. Por favor, reajusta tu búsqueda.
Nombre Id
--------------------
Neovim Neovim.Neovim
Neovim Neovim.Neovim
> winget uninstall --id Neovim.Neovim -v 0.9.5
No se encontró ningún paquete que coincida con los criterios de entrada.
```

### Environment

```shell
> winget --info
Administrador de paquetes de Windows v1.7.11261
Copyright (c) Microsoft Corporation. Todos los derechos reservados.

Windows: Windows.Desktop v10.0.22631.3593
Arquitectura del sistema: X64
Paquete: Microsoft.DesktopAppInstaller v1.22.11261.0

Directorios de WinGet
----------------------------------------------------------------------------------------------------------------------------------------
Registros                                   %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
Configuración de usuario                    %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json
Directorio de vínculos portátiles (usuario) %LOCALAPPDATA%\Microsoft\WinGet\Links
Directorio de vínculos portátiles (equipo)  C:\Program Files\WinGet\Links
Raíz de paquete portátil (usuario)          %LOCALAPPDATA%\Microsoft\WinGet\Packages
Raíz de paquete portátil                    C:\Program Files\WinGet\Packages
Raíz de paquete portátil (x86)              C:\Program Files (x86)\WinGet\Packages
Descargas del instalador                    %USERPROFILE%\Downloads

Vínculos
-----------------------------------------------------------------------------------
Declaración de privacidad   https://aka.ms/winget-privacy
Contrato de licencia        https://aka.ms/winget-license
Avisos de terceros          https://aka.ms/winget-3rdPartyNotice
Página principal            https://aka.ms/winget
Términos de Microsoft Store https://www.microsoft.com/en-us/storedocs/terms-of-sale

Configuración de administrador            Estado
-------------------------------------------------------
LocalManifestFiles                        Deshabilitado
BypassCertificatePinningForMicrosoftStore Deshabilitado
InstallerHashOverride                     Deshabilitado
LocalArchiveMalwareScanOverride           Deshabilitado
```

**Comment by stephengillie** [comment]
To separate this issue:
- The language switching should stay in this repo.
- The `Neovim.Neovim` issue should be moved to the [winget-pkgs](https://github.com/microsoft/winget-pkgs) repo.

I'm not sure of the best way to do this. Another Issue might be needed.

**Comment by mdanish-kh** [comment]
> I've run `winget upgrade --all` and it has installed the new version of Neovim (0.10.0) in parallel to the older one (0.9.5). Now I cannot uninstall any version even using the `-v` argument for `winget uninstall`.

@jfernandz Can you try out the [1.9-preview version of WinGet](https://github.com/microsoft/winget-cli/releases/tag/v1.9.1551-preview) and enable the [side-by-side experimental feature](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#experimental-features)? The preview version has better support for side-by-side installations. You would be able to do `winget uninstall --id Neovim.Neovim --version 0.9.5` to unblock this scenario.

Otherwise, you'll have to manually uninstall the old version from `appwiz.cpl` / Apps&Features

**Comment by mdanish-kh** [comment]
[Policy] Issue-Bug
[Policy] Side-by-Side

**Comment by jfernandz** [comment]
well, the manual method (`appwiz.cpl`) was able to uninstall Neovim `0.9.5`. 

`winget uninstall --id Neovim.Neovim --version 0.9.5` didn't work, but I've not tried the 1.9-preview version of WinGet 😢

maybe I should install it to check it the language issue persists; as Neovim issue is more appropriate for [winget-pkgs](https://github.com/microsoft/winget-pkgs)

**Comment by jfernandz** [comment]
`1.9-preview` version doesn't solve the language issue, btw 😞

**Comment by denelon** [comment]
```PowerShell
PS C:\Users\denelon> winget install Neovim.Neovim --version 0.9.5
Found Neovim [Neovim.Neovim] Version 0.9.5
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
This package requires the following dependencies:
  - Packages
      Microsoft.VCRedist.2015+.x64
(1/1) Found Microsoft Visual C++ 2015-2022 Redistributable (x64) [Microsoft.VCRedist.2015+.x64] Version 14.40.33810.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://download.visualstudio.microsoft.com/download/pr/1754ea58-11a6-44ab-a262-696e194ce543/3642E3F95D50CC193E4B5A0B0FFBF7FE2C08801517758B4C8AEB7105A091208A/VC_redist.x64.exe
  ██████████████████████████████  24.2 MB / 24.2 MB
Successfully verified installer hash
Starting package install...
Successfully installed

Downloading https://github.com/neovim/neovim/releases/download/v0.9.5/nvim-win64.msi
  ██████████████████████████████  40.4 MB / 40.4 MB
Successfully verified installer hash
Starting package install...
Successfully installed
PS C:\Users\denelon> winget upgrade Neovim.Neovim
Found Neovim [Neovim.Neovim] Version 0.10.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
This package requires the following dependencies:
  - Packages
      Microsoft.VCRedist.2015+.x64
Downloading https://github.com/neovim/neovim/releases/download/v0.10.0/nvim-win64.msi
  ██████████████████████████████  12.3 MB / 12.3 MB
Successfully verified installer hash
Starting package install...
Successfully installed
PS C:\Users\denelon> winget list Neovim.Neovim
Name   Id            Version Source
------------------------------------
Neovim Neovim.Neovim 0.10.0  winget
Neovim Neovim.Neovim 0.9.5   winget
PS C:\Users\denelon> winget uninstall Neovim.Neovim --version 0.9.5
Found Neovim [Neovim.Neovim]
Starting package uninstall...
Successfully uninstalled
PS C:\Users\denelon> winget list Neovim.Neovim
Name   Id            Version Source
------------------------------------
Neovim Neovim.Neovim 0.10.0  winget
PS C:\Users\denelon>
```

**Comment by denelon** [comment]
What do you have in your Preferred languages?

![image](https://github.com/microsoft/winget-cli/assets/61799811/5d6c1b14-e372-49e4-a020-be66c951ca19)

**Comment by denelon** [comment]
I'm going to close this issue since side-by-side is resolved, but if you're still having trouble with the locale, please file a new issue. Feel free to @ mention me in that issue.
